### PR TITLE
Make SigningOptions a property rather than passing explicitely to methods

### PR DESCRIPTION
**Pull request by NicolasDorier**
This allow us to get rid of the method overload explosion, and prevent foot shooting.

### CODE REVIEW

File: NBitcoin.TestFramework/WellknownNodeDownloadData.cs [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+using System.Linq;
 
 namespace NBitcoin.Tests
 {
 	public partial class NodeDownloadData
 	{
 		public bool SupportCookieFile { get; set; } = true;
 
-		public class BitcoinNodeDownloadData
+		public class NodeDownloadDataBase
```
Comment (knocte) [inline NBitcoin.TestFramework/WellknownNodeDownloadData.cs:10]: this seems unrelated to the PR, pushed by mistake?

Comment (NicolasDorier) [reply NBitcoin.TestFramework/WellknownNodeDownloadData.cs:10]: Unrelated to this PR, I did a separate commit.

Comment (knocte) [reply NBitcoin.TestFramework/WellknownNodeDownloadData.cs:10]: but the NodeDownload commit had signing-related things too xD
